contracts/ERC721M.sol

```diff
@@ -448,11 +457,23 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
         }
 
         if (_mintCurrency != address(0)) {
+            // ERC20 mint payment
             IERC20(_mintCurrency).safeTransferFrom(
```

Comment:
we should just use transferFrom for both calls.

safeTransferFrom is meant as a check for smart contracts that can support erc20 token transfers. Since we're a launchpad i think its fair that we shouldnt support smart contracts minting from our launchpad, and shouldnt care if they fail to accept erc20 tokens.

for MINT_FEE_RECEIVER, if its a contract we own then we can safely not use safeTransferFrom as long as we know it can accept erc20 tokens. if its a EOA address we should just use transferFrom.

calling safeTransferFrom also has extra gas over transferFrom

Reply:
+1 agree. Fixed.